### PR TITLE
udev: re-introduce symlinks for loopback block device

### PR DESCRIPTION
**Pull request by yuwata**
This effectively reverts #28476.

### CODE REVIEW

**Comment by bluca** [comment]
@karelzak what do we need to make this work?

**Comment by DaanDeMeyer** [comment]
Can't we just name this `/dev/loop-by-ref`, ... to avoid the conflict?

**Comment by karelzak** [comment]
The problem is the directory name, if there is /dev/loop/ directory then losetup assumes loopN device nodes are in the directory rather than in the /dev.

So, if you use another name (e.g. /dev/loop-by-ref) and another location  (/dev/disk/by-ref) then it will be enough.

**Comment by yuwata** [comment]
Updated. Changed /dev/loop/ -> /dev/disk/. PTAL.

**Review by bluca — APPROVED** [review]
Thanks looks good to me, but would like an explicit blessing from @karelzak too

**Comment by karelzak** [comment]
The location /dev/disk/by-ref/ is the best :-) Thanks.

File: NEWS [hunk 1]
```diff
@@ -33,6 +33,23 @@ CHANGES WITH 255 in spe:
           by default when combined with --scope, will be changed in a future
           release to be enabled by default.
 
+        Device Management:
+
+        * udev will now create symlinks to loopback block devices in the
+          /dev/disk/by-ref/ directory that are based on the .lo_file_name
```
Comment (poettering) [inline NEWS]: hmm, /dev/disk/ is a bit generic for some super-vague term just "ref". this names should really carry some reference to "loop".

Maybe /dev/disk/by-loop-ref/? Or /dev/disk/loop/by-ref/? 

Kinda leaning forwards the former, i.e. /dev/disk/by-loop-ref/ + /dev/disk/by-loop-inode/?

Comment (poettering) [reply NEWS]: The switch to losetup is named --loop-ref=, hence I theink /dev/disk/by-loop-ref/ would be the most obvious choice, to directly reuse that naming in the symlink.

**Comment by yuwata** [comment]
@poettering Thank you for the suggestion. The directories are changed again as suggested. PTAL.

**Comment by poettering** [comment]
thanks!

**Pull request merged**